kotlinpoet/src/main/java/com/squareup/kotlinpoet/Util.kt

```diff
@@ -151,7 +152,13 @@ internal fun String.escapeIfKeyword() = if (isKeyword) "`$this`" else this
 internal fun String.escapeIfNotJavaIdentifier() =
   if (!Character.isJavaIdentifierStart(first()) || drop(1).any { !Character.isJavaIdentifierPart(it) }) "`$this`" else this
 
-internal fun String.escapeIfNecessary() = escapeIfNotJavaIdentifier().escapeIfKeyword()
+internal fun String.escapeIfNecessary() = escapeIfNotJavaIdentifier().escapeIfKeyword().failIfEscapeInvalid()
+
+internal fun String.failIfEscapeInvalid() =
+  if (any { it in illegalCharactersToEscape })
```

Comment:
nit: Let's use `{}` with this if-else block.

Reply:
I changed it to a require. After adding a few braces intellij suggested it and I think it's better this way.